python/adjoint/objective.py

```diff
@@ -483,3 +495,127 @@ def __call__(self):
         self.ldos_scale = self.sim.ldos_scale
         self.ldos_Jdata = self.sim.ldos_Jdata
         return np.array(self._eval)
+    
+class PoyntingFlux(ObjectiveQuantity):
+    """A frequency-dependent Poynting Flux adjoint source.
+    Attributes:
+        volume: The volume over which the Poynting Flux is calculated.
+        This function currently only works for monitors with a defined
+        normal vector (e.g. planes in 3d or lines in 2d). User supplied
+        normal vectors may be implemented in the future. It also only
+        works with monitors aligned to a coordinate direction.
+        decimation_factor: Whether to skip points in the time series every 
+        decimation_factor timesteps. See "add_dft_fields" documentation.
+        The biggest warning there is to be careful to avoid aliasing if
+        the fields vary quickly in time.
+        Note on yee_grid: For the Poynting Flux to work, H and E components
+        must lie at the same points. Therefore, the Yee grid will always be false.
+    """
+
+    def __init__(self, sim,volume, decimation_factor=0):
```

Comment:
You should have a parameter `scale`, just like we do with the `PoyntingFlux` class.

Reply:
Added a few commits ago! Thank you for the feedback.